docs/pages/guides/onchain-indices.mdx

```diff
@@ -0,0 +1,412 @@
+
+   <CollapseCode>
+
+```solidity filename="MoveSystem.sol" copy showLineNumbers {6,12-13}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { setPosition } from "./setPosition.sol";
+import { PlayersAtPosition } from "../codegen/index.sol";
+
+contract MoveSystem is System {
+  function move(int32 x, int32 y) public {
+    address player = _msgSender();
+
+    bool positionIsEmpty = PlayersAtPosition.length(x, y) == 0;
+    require(positionIsEmpty, "another player is at this position");
+
+    setPosition(player, x, y);
+  }
+}
+```
+
+   </CollapseCode>
+
+#### `PositionIsOccupied`
+
+Having a list of players at each position is useful, but ultimately we only want to know if _some_ player is at the position, not their specific addresses. In addition, the "occupied" check ensures that at most one player can be at a given position.
+
+Therefore we can take advantage of managing this in our own contracts and write a simpler index.
+
+This time, we declare a table called `PositionIsOccupied`:
+
+   <CollapseCode>
+```ts filename="mud.config.ts" copy showLineNumbers {14-22}
+import { mudConfig } from "@latticexyz/world/register";
+
+export default mudConfig({
+  tables: {
+    Position: {
+      keySchema: {
+        player: "address",
+      },
+      valueSchema: { 
+        x: "int32", 
+        y: "int32" 
+      },
+    },
+    PositionIsOccupied: {
+      keySchema: { 
+        x: "int32", 
+        y: "int32"
+      },
+      valueSchema: { 
+        isOccupied: "bool" 
+      },
+    }
+  }
+});
+```
+
+  </CollapseCode>
+
+To maintain this index when players move, we flag the previous position as empty and the new one as occupied.
+
+```solidity filename="setPosition.sol" copy showLineNumbers
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { Position, PositionData, PositionIsOccupied } from "../codegen/index.sol";
+
+function setPosition(address player, int32 x, int32 y) {
+  // Cache the player's previous position
+  PositionData memory previousPosition = Position.get(player);
+
+  Position.set(player, x, y);
+
+  PositionIsOccupied.set(previousPosition.x, previousPosition.y, false);
+  PositionIsOccupied.set(x, y, true);
+}
+```
+
+Finally, to check whether the position is occupied, we simply read the index.
+
+   <CollapseCode>
+
+```solidity filename="MoveSystem.sol" copy showLineNumbers {5,11}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { Position, PositionIsOccupied } from "../codegen/index.sol";
+
+contract MoveSystem is System {
+  function move(int32 x, int32 y) public {
+    address player = _msgSender();
+
+    bool positionIsEmpty = !PositionIsOccupied.get(x, y);
+    require(positionIsEmpty, "another player is at this position");
+
+    Position.set(player, x, y);
+  }
+}
+```
+
+   </CollapseCode>
+
```

Comment:
Add a note here that store hooks (linking to /store/store-hooks) can be used to make sure nobody sets a position without changing the index.